File: Gerrymandering/Gerry.java
import java.util.Scanner;
import java.text.DecimalFormat;

public class Gerry {
  public static void main(String[] args) {
    Scanner scan = new Scanner(System.in);
    int inputs = scan.nextInt();
    int districts = scan.nextInt();
    int[][] votes = new int[districts][2];
    for (int i = 0; i < districts; i++) {
      votes[i][0] = 0;
      votes[i][1] = 0;
    }
    int totalVs = 0;
    for (int i = 0; i < inputs; i++) {
      int x = scan.nextInt();
      votes[x - 1][0] += scan.nextInt();
      votes[x - 1][1] += scan.nextInt();
    }
    scan.close();
    for (int i = 0; i < districts; i++) {
      totalVs += votes[i][0] + votes[i][1];
    }
    int wa = 0;
    int wb = 0;
    for (int i = 0; i < districts; i++) {
      if (votes[i][0] > votes[i][1]) {
        System.out.print("A ");
        int y = votes[i][0] - ((votes[i][0] + votes[i][1]) / 2 + 1);
        wa += y;
        System.out.print(y + " ");
        System.out.println(votes[i][1]);
        wb += votes[i][1];
      } else {
        System.out.print("B ");
        System.out.print(votes[i][0] + " ");
        wa += votes[i][0];
        int y = votes[i][1] - ((votes[i][0] + votes[i][1]) / 2 + 1);
        wb += y;
        System.out.println(y);
      }
    }
    double d = Math.abs(wa - wb) / (double) totalVs;
    System.out.println(d);
  }
}

File: Gerrymandering/Gerry.cpp
#include <iostream>
#include <vector>
using namespace std;
int main(int argc, char const *argv[])
{
  int inputs, districts, totalVs = 0, wa = 0, wb = 0, d, a, b;
  cin >> inputs >> districts;
  vector< vector<int> > votes;
  for (int i = 0; i < districts; i++)
    votes.push_back(vector<int>());
  for (int i = 0; i < inputs; i++)
  {
    cin >> d >> a >> b;
    cout << d << " " << a << " " << b << "\n";
    votes.at(d - 1).push_back(a);
    votes.at(d - 1).push_back(b);
    totalVs += a + b;
  }
  int y;
  for (int i = 0; i < districts; i++)
  {
    if (votes.at(i).at(0) > votes.at(i).at(1))
    {
      cout << "A ";
      y = votes.at(i).at(0) - ((votes.at(i).at(0) + votes.at(i).at(1)) / 2 + 1);
      wa += y;
      cout << y << " " << votes.at(i).at(1) << "\n";
      wb += votes.at(i).at(1);
    }
    else
    {
      cout << "B " << votes.at(i).at(0) << " ";
      wa += votes.at(i).at(0);
      y = votes.at(i).at(1) - ((votes.at(i).at(0) + votes.at(i).at(1)) / 2 + 1);
      wb += y;
      cout << y << "\n";
    }
  }
  cout << ((double)(wa - wb >= 0 ? wa - wb : -(wa - wb)) / (double)totalVs) << "\n";
  return 0;
}
